### PR TITLE
generic value types

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by danielnorberg** [comment]
@mbruggmann PTAL

File: processor/src/main/java/io/norberg/automatter/processor/Descriptor.java [hunk 1]
```diff
@@ -107,6 +129,18 @@ public boolean isPublic() {
...
 
+  public boolean isGeneric() {
+    return this.isGeneric;
+  }
+
+  public List<? extends TypeMirror> typeArguments() {
+    return typeArguments;
+  }
+
+  public String typeParameterization() {
```
Comment (mbruggmann) [inline processor/src/main/java/io/norberg/automatter/processor/Descriptor.java]: Is this unused?

**Comment by mbruggmann** [comment]
Minor nitpick, otherwise :+1:

File: processor/src/main/java/io/norberg/automatter/processor/Descriptor.java (hunk 1 above)
Comment (danielnorberg) [inline processor/src/main/java/io/norberg/automatter/processor/Descriptor.java]: good catch!

**Comment by danielnorberg** [comment]
@mbruggmann :beers:

**Comment by mbruggmann** [comment]
:+1:

**Comment by rouzwawi** [comment]
Looks good overall. Haven't looked much at the implementation though.

Might be good to add some tests that cover multiple levels of wildcards like:

```java
public interface GenericMultiple<
    T1,
    T2 extends Collection<? extends T1>,

  T1 field1();
  T2 field2();
}
```

Mainly to cover code that uses the builder method `field1(...)` with variables of various types that should lie within the bounds.

With `T1 == Number`:

* `List<Integer>`
* `List<? extends Integer>`
* `List<Number>`
* `List<? extends Number>`

**Pull request merged**